.github/workflows/diff-report.yml

```diff
@@ -227,19 +227,17 @@ jobs:
       - name: Set output
         id: out
         run: |
-          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
-          {
-            echo "message<<$EOF"
-            cat message
-            echo "$EOF"
-          } >> "$GITHUB_OUTPUT"
+          cd checkstyle
+          ./.ci/append-to-github-output.sh "message" "$(cat ../message)"
 
   # should be always last step
   send_message:
     runs-on: ubuntu-latest
     needs: [ parse_body, make_report ]
     if: failure() || success()
     steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```

Comment:
Does it make sense to use `fetch-depth: 0`

And align affected scripts to use v3 of action.

Reply:
Opened PR at https://github.com/checkstyle/checkstyle/pull/12960.